### PR TITLE
Fixes #142 Input username and reponame instead of URL

### LINKED ISSUE #142: Ask user to input username and repository name instead of URL
### This is a
- [ ] Bug report
- [x] Feature request

### Description

The user is asked to input a url of their repository in the webapp. Apparently, there can be different ways of inputting a URL. Asking the user to input username and repository name would be simpler for the user to input. Also, it will be simpler in our end to generate a URL from it.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Related Issue

issue: #142 

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
@imujjwal96 if we keep input box like this, Then yaydoc will get restricted to GitHub

**Comment by imujjwal96** [comment]
https://github.com/fossasia/labs.fossasia.org/issues/210

The description of the Project mentions that 
> The goal of this project is to develop and implement a number of tools, that help us to build dev websites fully automatically from GitHub repositories that provide full features including search

**Comment by sch00lb0y** [comment]
@imujjwal96 I agree, but then generating documentation from other service providers like bitbucket, gitlab is not bad. Have checkbox saying github. if it's github show the two input box.

**Comment by imujjwal96** [comment]
My concern is that if we do this, then it would be expected that we deploy to these services too. I don't think we'll be able to manage this much.

I personally think, we should first focus on Github, make sure that everything works perfectly for it, and then if we have time, we could extend it to other services.

Its better if we create a system that is bug-free even though it has less features.

File: backend/generator.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ exports.executeScript = function (socket, formData) {
     return false;
   }
   var email = formData.email;
-  var gitUrl = formData.gitUrl;
+  var gitUrl = "https://github.com/" + formData.username + "/" + formData.reponame + ".git";
```
Comment (pri22296) [inline backend/generator.js]: Remove gitUrl entirely. instead pass reponame and username to the script and build the url there

Comment (imujjwal96) [inline backend/generator.js]: What about the CI part?

Comment (pri22296) [inline backend/generator.js]: could you elaborate?

Comment (imujjwal96) [inline backend/generator.js]: Will Travis handle that automatically? Cuz if not here, we'll have to form a URL inside the generate script (to clone the repo)

Comment (pri22296) [inline backend/generator.js]: The gitURL parameter is only used by the WEBUI part of the script. So it shouldn't affect the travis part. Yeah you would have to build the url there

Comment (imujjwal96) [inline backend/generator.js]: Building there or here, whats the difference?

Comment (pri22296) [inline backend/generator.js]: In the webUI, we are using username, reponame in some places and gitURL in others. using username, reponame seperately in the webUI would make it consistent.

**Comment by imujjwal96** [comment]
@pri22296 updated

File: public/scripts/form.js [hunk 2]
```diff
@@ -33,29 +44,14 @@ $(function () {
...
-    $('#notification-container').css("opacity", "1");
-    setTimeout(function() {
-      $('#notification-container').css("visibility", "hidden");
-      $('#notification-container').css("opacity", "0");
-    }, 5000)
-    $('#btnDeploy').css("display", "inline");
-    $('#btnDeploy').attr("href", '/github?email='+data.email+'&uniqueId='+data.uniqueId+'&gitURL='+data.gitUrl);
+    $('#btnDownload').css("display", "inline").attr("href", "/download/" + data.email +"/" + data.uniqueId);
+    $('#btnPreview').css("display", "inline").attr("href", "/preview/" + data.email +"/" + data.uniqueId + "_preview");
+    $('#btnDeploy').css("display", "inline").attr("href", '/github?email='+data.email+'&uniqueId='+data.uniqueId+'&gitURL='+data.gitUrl);
```
Comment (pri22296) [inline public/scripts/form.js]: remove giturl from here too

File: routes/index.js [hunk 3]
```diff
@@ -15,21 +16,21 @@ router.get('/download/:email/:uniqueId', function (req, res, next) {
 
 router.get("/github", function (req, res, next) {
   req.session.uniqueId = req.query.uniqueId;
-  req.session.email = req.query.email
-  req.session.gitURL = req.query.gitURL
+  req.session.email = req.query.email;
+  req.session.gitURL = req.query.gitURL;
```
Comment (pri22296) [inline routes/index.js]: remove giturl from here too

**Comment by imujjwal96** [comment]
@pri22296 updated

**Comment by pri22296** [comment]
@imujjwal96 Please resolve merge conflicts.

**Comment by imujjwal96** [comment]
Merge conflicts resolved.

**Comment by mariobehling** [comment]
Why is this necessary? We do we need to make a distinction between users? And where are organizations? We already have a simple identification. Let's not focus on cosmetic issues and rather get everything to work, e.g. when will you start to support multiple repositories? What is the way to go here? 

We need a loklak main page and then subsites for each repositories. Please open issues and focus on real features. 

As this is diverting us from focusing on features I am closing this PR. It is starting to make a separation between users and organizations that is not necessary. Instead of having to fill in one form field, I would have to fill in two form fields. This is not an improvement.

**Pull request closed without merging**